snowplow-tracker/src/androidTest/java/com/snowplowanalytics/snowplow/tracker/TrackerWebViewInterfaceV2Test.kt

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) 2015-present Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.snowplow.tracker
+
+import android.content.Context
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import com.snowplowanalytics.core.constants.Parameters
+import com.snowplowanalytics.core.constants.TrackerConstants
+import com.snowplowanalytics.core.emitter.Executor
+import com.snowplowanalytics.core.tracker.TrackerWebViewInterfaceV2
+import com.snowplowanalytics.snowplow.Snowplow.createTracker
+import com.snowplowanalytics.snowplow.Snowplow.removeAllTrackers
+import com.snowplowanalytics.snowplow.configuration.NetworkConfiguration
+import com.snowplowanalytics.snowplow.configuration.PluginConfiguration
+import com.snowplowanalytics.snowplow.configuration.TrackerConfiguration
+import com.snowplowanalytics.snowplow.controller.TrackerController
+import com.snowplowanalytics.snowplow.network.HttpMethod
+import org.json.JSONException
+import org.json.JSONObject
+import org.junit.After
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class TrackerWebViewInterfaceV2Test {
+    private var webInterface: TrackerWebViewInterfaceV2? = null
+    private var networkConnection = MockNetworkConnection(HttpMethod.GET, 200)
```

Comment:
This is the old annoying way to test events that requires long sleeps for events to arrive in the requests. More recently we have been using the `EventSink` plugin where you can also check the payload properties without having to wait for the emitter to send the events (which we don't need to test in this test case).

Reply:
On the other hand, that would allow you to test the wrapping of the self-describing data that happens in the `TrackerEvent`, but maybe that is worth a separate test file for that class and functionality?

Reply:
I originally set up the tests using a filter plugin, but I changed it so that the `TrackerEvent` processing would be included.

Actually there's already a mix because the `addsEventNameAndSchemaForInspection()` test uses a filter (didn't realise we had a whole class for this, nice).

I could change it so the `tracksEventWithCorrectTracker()` test also uses afterTrack filter/`EventSink` - don't think it's worth separating out that first test (which relies on `TrackerEvent` stuff) into a new class. Unless you think it's worth it for neatness?

Reply:
That makes sense! Yeah, if using the network connection is useful, we can keep that, don't have to complicate it. We can keep the first test using that.